airflow/models/dag.py

```diff
@@ -3518,7 +3541,7 @@ def deactivate_deleted_dags(cls, alive_dag_filelocs: list[str], session=NEW_SESS
         """
         log.debug("Deactivating DAGs (for which DAG files are deleted) from %s table ", cls.__tablename__)
 
-        dag_models = session.query(cls).all()
+        dag_models = session.scalars(select(cls)).all()
         for dag_model in dag_models:
```

Comment:
```suggestion
        for dag_model in session.scalars(select(cls)):
```

(I wonder if this can actually be rewritten with `update`)

Reply:
Maybe in a different PR